packages/algob/src/lib/ssc.ts

```diff
@@ -1,23 +1,84 @@
 import tx, { Account as AccountSDK, ConfirmedTxInfo, decodeAddress } from "algosdk";
 
 import { AlgobDeployer, SSCOptionalFlags, TxParams } from "../types";
+import { MAX_UINT64, MIN_UINT64 } from "./constants";
 import { encodeNote, mkTxParams } from "./tx";
 
+export const reDigit = /^\d+$/;
+
+// verify n is an unsigned 64 bit integer
+function assertUint64 (n: bigint): void {
+  if (n < MIN_UINT64 || n > MAX_UINT64) {
+    throw new Error(`Invalid uint64 ${n}`);
+  }
+}
+
 /**
- * Converts string to bytes.
- * @param s string
+ * Converts 64 bit unsigned integer to bytes in big endian.
  */
-export function stringToBytes (s: string): Uint8Array {
-  return new Uint8Array(Buffer.from(s));
+export function uint64ToBigEndian (x: number | bigint): Uint8Array {
+  const bytes = new Uint8Array(8);
+  let i = 0;
+  x = BigInt(x); // use x as bigint internally to support upto uint64
+  assertUint64(x);
+  while (x) {
+    bytes[i++] = Number(x % 256n);
+    x /= 256n;
+  }
+  return bytes.reverse();
```

Comment:
If you start with `i=7` and use `bytes[i--] = Number(x % 256n)` then you don't need to call `bytes.reverse()`.

Reply:
Thanks for suggestion!